src/explorer.ts

```diff
@@ -161,6 +160,10 @@ class KubernetesResourceFolder extends KubernetesFolder {
             return [new DummyObject("Error")];
         }
         return childrenLines.result.map((line) => {
+            if (this.kind.abbreviation==="pod"){
+            const bits = line.replace(/\s+/g,'|').split('|');
```

Comment:
I'm a little bit iffy about using regex to do data massaging for tabular formats... could you change the above

can you change the getChildren function to get data from kubectl as json? the Kubectl interface offers an `asJson` function. here's an example of its usage:

`const shellResult = await kubectl.asJson<any>("config view -o json");`

Reply:
Sounds good! Will try `asJson`

Reply:
thinking about how to switch to `asJson` and get the same data as with `asLines`. I can retrieve the `ID` but getting an consolidated pod status might require additional parsing and aggregation/sorting of the JSON output as with JSON the output contains individual `containerStatuses` in the pod. Let me know if you have any pointers regarding the approach.

Reply:
@shreyu86 I think you'd want `status.phase` from the resulting json.

Reply:
Tried to refactor with `status.phase` but an issue more of a `kubectl` one that I ran into was that `status.phase` (in my case `status.phase` was always `running` even if pod was crashing ) was that "The phase is not intended to be a comprehensive rollup of observations of Container or Pod state, nor is it intended to be a comprehensive state machine." and hence preferred the current output as it performs an rollup. What I might try to do is write a function to perform the rollup in an followup PR after doing some prototyping.